### PR TITLE
Restrict access to Legacy Gradle plugins

### PR DESCRIPTION
**Pull request by breskeby**
Restrict the usage of Legacy Gradle plugins to the current usage. Adding legacy projects to new projects will make the build fail.

### CODE REVIEW

**Review by mark-vieira — COMMENTED** [review]
There are projects in here that are

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/DeprecationService.java [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+    public DeprecationService() {
+        addLegacyRestTestBasePluginUsageLimitations();
+    }
+
+    private static void addLegacyRestTestBasePluginUsageLimitations() {
+        usageWhitelist.put(LegacyRestTestBasePlugin.class, ":docs");
+        usageWhitelist.put(LegacyRestTestBasePlugin.class, ":rest-api-spec");
+        usageWhitelist.put(LegacyRestTestBasePlugin.class, ":distribution:docker");
+        usageWhitelist.put(LegacyRestTestBasePlugin.class, ":modules:aggregations");
```
[the remark this answers is not in the document]
Comment (breskeby) [reply build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/DeprecationService.java:32]: Ah. the yaml-rest-test-combat plugin is using the legacy base plugin. I *think* this is not right. Checking

[the remark this answers is not in the document]
Comment (mark-vieira) [reply build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/DeprecationService.java:32]: Yeah, that's probably an oversight.

[the remark this answers is not in the document]
Comment (breskeby) [reply build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/DeprecationService.java]: cleaned that up

**Review by mark-vieira — APPROVED** [review]

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/DeprecationService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mark-vieira) [reply build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/DeprecationService.java]: Thanks!

**Pull request merged**